### PR TITLE
Fix/parser variable arity

### PR DESCRIPTION
**Pull request by clementdessoude**
Fix last parser identified bugs with variable Arity parameters (see #139)

### CODE REVIEW

File: packages/java-parser/src/productions/types-values-and-variables.js [hunk 1]
```diff
@@ -136,12 +136,15 @@ function defineRules($, t) {
...
     $.CONSUME(t.LSquare);
     $.CONSUME(t.RSquare);
-    $.MANY2(() => {
-      $.MANY3(() => {
-        $.SUBRULE2($.annotation);
-      });
-      $.CONSUME2(t.LSquare);
-      $.CONSUME2(t.RSquare);
+    $.MANY2({
+      GATE: () => $.BACKTRACK_LOOKAHEAD($.isDims),
```
Comment (bd82) [inline packages/java-parser/src/productions/types-values-and-variables.js]: this GATE seems too restrictive, what if it starts with annotation?

Comment (clementdessoude) [reply packages/java-parser/src/productions/types-values-and-variables.js]: I don't really understand: if it start with annotations, the Gate will be OK, as the `isDims` rule start with MANY(annotation). Or am I missing something? The name isDims is maybe a bit misleading

Comment (clementdessoude) [reply packages/java-parser/src/productions/types-values-and-variables.js]: I don't really understand: if it start with annotations, the Gate will be OK, as the `isDims` rule start with MANY(annotation). Or am I missing something? The name isDims is maybe a bit misleading

Comment (bd82) [reply packages/java-parser/src/productions/types-values-and-variables.js]: I missed that isDims also checks annotations

**Pull request merged**